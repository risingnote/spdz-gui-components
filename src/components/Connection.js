/**
 * Responsible for displaying SPDZ connection information based on props passed in.
 */
import React from 'react';
import { List } from 'immutable'

import ProxyStatus from './ProxyStatus'
import { generateProxyStatusList } from './ProxyStatusHelper'

const Connection = (props) => {
  const combinedComponentStyle = Object.assign(
    {
      padding: '0px 10px 5px 10px',
      border: '1px solid darkgrey',
      margin: '0 auto',
      textAlign: 'left'
    }, props.componentStyle)

  const combinedHeadingStyle = Object.assign(
    {
      fontSize: '16px',
      color: 'rgb(28, 118, 152)',
      marginRight: '10px'
    }, props.headerStyle)

  const proxyServers = (spdzProxyServerList, spdzProxyStatus) => {
    const proxyStatusForDisplay = generateProxyStatusList(spdzProxyServerList, spdzProxyStatus)
    return proxyStatusForDisplay.map( (proxyServer) => {
      return <ProxyStatus key={proxyServer.get('url')} status={proxyServer.get('status')} 
                url={proxyServer.get('url')} urlStyle={props.urlStyle} />
    })
  }

  return (
    <div style={combinedComponentStyle}>
      <p style={combinedHeadingStyle}>Spdz Proxy Servers</p>
      {proxyServers(props.spdzProxyServerList, props.spdzProxyStatus)}
    </div>
  )
} 

Connection.propTypes = {
  spdzProxyServerList: React.PropTypes.instanceOf(List).isRequired,
  spdzProxyStatus: React.PropTypes.array.isRequired,
  componentStyle: React.PropTypes.object,
  headerStyle: React.PropTypes.object,
  urlStyle: React.PropTypes.object
}

export default Connection