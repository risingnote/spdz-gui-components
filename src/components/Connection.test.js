import React from 'react'
import renderer from 'react-test-renderer'
import { List, Map } from 'immutable'
import { ProxyStatusCodes } from 'spdz-gui-lib' 

import Connection from './Connection'

const twoProxies = List.of(
  Map({
    url: "http://spdzProxy.one:4000"
  }),
  Map({
    url: "http://spdzProxy.two:4000"
  })
)

const status1Connected = [
  {
    id: 0,
    status: ProxyStatusCodes.Failure
  },
  {
    id: 1,
    status: ProxyStatusCodes.Connected
  }
]

describe('Setup display component rendering', () => {
  it('Renders as expected (compared to a snapshot) when passed 2 proxy connections', () => {

    const tree = renderer.create(
      <Connection spdzProxyServerList={twoProxies} spdzProxyStatus={status1Connected}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Renders as expected (compared to a snapshot) when passed no proxy connections', () => {

    const tree = renderer.create(
      <Connection spdzProxyServerList={List()} spdzProxyStatus={[]}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
