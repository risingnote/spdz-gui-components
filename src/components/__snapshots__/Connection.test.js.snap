// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup display component rendering Renders as expected (compared to a snapshot) when passed 2 proxy connections 1`] = `
<div
  style={
    Object {
      "border": "1px solid darkgrey",
      "margin": "0 auto",
      "padding": "0px 10px 5px 10px",
      "textAlign": "left",
    }
  }
>
  <p
    style={
      Object {
        "color": "rgb(28, 118, 152)",
        "marginRight": "10px",
      }
    }
  >
    Spdz Proxy Servers
  </p>
  <div
    style={
      Object {
        "display": "flex",
        "flexFlow": "row wrap",
        "margin": "0px 0px 10px 0px",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "lightcoral",
          "border": "1px solid black",
          "borderRadius": "50%",
          "width": "16px",
        }
      }
    />
    <div
      style={
        Object {
          "fontSize": "12px",
          "paddingLeft": "10px",
        }
      }
    >
      http://spdzProxy.one:4000
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexFlow": "row wrap",
        "margin": "0px 0px 10px 0px",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "lightgreen",
          "border": "1px solid black",
          "borderRadius": "50%",
          "width": "16px",
        }
      }
    />
    <div
      style={
        Object {
          "fontSize": "12px",
          "paddingLeft": "10px",
        }
      }
    >
      http://spdzProxy.two:4000
    </div>
  </div>
</div>
`;

exports[`Setup display component rendering Renders as expected (compared to a snapshot) when passed no proxy connections 1`] = `
<div
  style={
    Object {
      "border": "1px solid darkgrey",
      "margin": "0 auto",
      "padding": "0px 10px 5px 10px",
      "textAlign": "left",
    }
  }
>
  <p
    style={
      Object {
        "color": "rgb(28, 118, 152)",
        "marginRight": "10px",
      }
    }
  >
    Spdz Proxy Servers
  </p>
</div>
`;
